### DISCUSSION TITLE
Error when executing service method 'StartExecution'. System.InvalidOperationException: Cannot locate Gauge.CSharp.Lib.ScreenshotFilesCollector in Gauge.CSharp.Lib ---> System.InvalidOperationException: Sequence contains no matching

### DISCUSSION THREAD

**Discussion by pratapkuchipudi** [getgauge/gauge#1830] [General]
I have recently joined with an organization where they are using gauge
and they have an existing automation project

i installed everything and tried to access/run the existing suite but i am not able to succeeded  in it 
for all other employees it is working fine
but i am getting the following error
Error when executing service method 'StartExecution'. System.InvalidOperationException: Cannot locate Gauge.CSharp.Lib.ScreenshotFilesCollector in Gauge.CSharp.Lib ---> System.InvalidOperationException: Sequence contains no matching
[log.txt](https://github.com/getgauge/gauge/files/5751333/log.txt)

**Comment by pratapkuchipudi** [comment]
Team, 
any one there ...
it is bit urgent and critical for me

**Reply by sriv** [reply]
What's the version of `Gauge.CSharp.Lib` in the project?

Are.you able to run a fresh gauge project?

**Reply by pratapkuchipudi** [reply]
No i tried with old one and new one
for all it is the same

gauge --version
Gauge version: 1.1.6
Commit Hash: 3af456f

Plugins
-------
dotnet (0.3.0)
html-report (4.0.12)
java (0.7.13)
python (0.3.14)
screenshot (0.0.1)
spectacle (0.1.4)
xml-report (0.2.3)

**Reply by sriv** [reply]
Check your `csproj` for version of Gauge.CSharp.Lib

**Reply by pratapkuchipudi** [reply]
here are the details in csproj file

<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="FluentAssertions" Version="4.19.4"/>
    <PackageReference Include="Gauge.CSharp.Lib" Version="0.7.2"/>
    <PackageReference Include="Selenium.WebDriver" Version="3.141.0"/>
    <PackageReference Include="Selenium.Support" Version="3.141.0"/>
    <PackageReference Include="DotNetSeleniumExtras.WaitHelpers" Version="3.11.0"/>
    <PackageReference Include="Selenium.Chrome.WebDriver" Version="77.0.0"/>
    <PackageReference Include="Selenium.Firefox.WebDriver" Version="0.26.0"/>
    <PackageReference Include="Selenium.WebDriver.IEDriver" Version="3.150.1"/>
    <PackageReference Include="CsQuery" Version="1.3.4"/>
  </ItemGroup>
</Project>

**Comment by pratapkuchipudi** [comment]
please find the details in csproj file 
 PackageReference Include="Gauge.CSharp.Lib" Version="0.7.2"/

and what is the latest csharp.lib file version we have

**Reply by sriv** [reply]
Try setting the version to 0.8.1

**Reply by sriv** [reply]
(Which is the latest version)

**Comment by pratapkuchipudi** [comment]
Thanks Srikanth,
updated to version 0.8.1
now getting another error
nfo: Microsoft.AspNetCore.Routing.EndpointMiddleware[0]
Executing endpoint 'gRPC - /gauge.messages.Runner/InitializeSuiteDataStore'
info
: Grpc.AspNetCore.Server.ServerCallHandler[7]
Error status code 'Unimplemented' raised.
Grpc.Core.RpcException: Status(StatusCode="Unimplemented", Detail="")
at Gauge.Messages.Runner.RunnerBase.InitializeSuiteDataStore(SuiteDataStoreInitRequest request, ServerCallContext context)
at Grpc.Shared.Server.UnaryServerMethodInvoker`3.Invoke(HttpContext httpContext, ServerCallContext serverCallContext, TRequest request)
--- End of stack trace from previous location where exception was thrown ---
at Grpc.AspNetCore.Server.Internal.CallHandlers.UnaryServerCallHandler`3.HandleCallAsyncCore(HttpContext httpContext, HttpContextServerCallContext serverCallContext)
at Grpc.AspNetCore.Server.Internal.CallHandlers.ServerCallHandlerBase`3.<HandleCallAsync>g__AwaitHandleCall|8_0(HttpContextServerCallContext serverCallContext, Method`2 method, Task handleCall)
info
: Microsoft.AspNetCore.Routing.EndpointMiddleware[1]
Executed endpoint 'gRPC - /gauge.messages.Runner/InitializeSuiteDataStore'
info
: Microsoft.AspNetCore.Hosting.Diagnostics[2]
Request finished in 14.6496ms 200 application/grpc
  failed to initialize suite datastore. Error: 
Unable to connect to plugin Html Report 4.0.12. rpc error: code = Unavailable desc = transport is closing

Failed to kill plugin Html Report 4.0.12. TerminateProcess: Access is denied.

info
: Microsoft.AspNetCore.Hosting.Diagnostics[1]

**Reply by sriv** [reply]
I'll need to study this in my computer, but in the meantime, try installing gauge-dotnet 0.2.0.

You can run

```
gauge uninstall dotnet
gauge install dotnet --version 0.2.0
```

**Reply by sriv** [reply]
I am on 

```
Gauge version: 1.1.6
Commit Hash: 3af456f

Plugins
-------
dotnet (0.3.0)
flash (0.0.2)
go (0.1.4)
html-report (4.0.12)
java (0.7.10)
js (2.3.12)
json-report (0.3.3)
python (0.3.10)
screenshot (0.0.1)
spectacle (0.1.4)
xml-report (0.2.3)

```

I am able to create a new project and run it fine:

```
gauge init dotnet
gauge run specs
...
...
Specifications:	1 executed	1 passed	0 failed	0 skipped
Scenarios:	2 executed	2 passed	0 failed	0 skipped

Total time taken: 235ms
```

**Reply by sriv** [reply]
What happens if you remove `html-report` from your manifest.json? i.e. leave `plugins` as an empty array and execute the project?

**Comment by pratapkuchipudi** [comment]
Is there any personal support/customer care support for gauge new users...????
kindly direct us to it...


if possible send us your mail id will share our project to you ...........

**Reply by sriv** [reply]
Not that I know of.

**Reply by pratapkuchipudi** [reply]
if possible send us your mail id will share our project to you ...........

**Reply by sriv** [reply]
You're welcome to post questions here, the maintainers or other users can pitch in. Please ensure you share enough information, the speed and probability of a response is directly proportional to the specific information made available.

**Reply by sriv** [reply]
I'm afraid I cannot accept any project that is not publicly shareable. It may lead to legal issues for not just us, but potentially our respective employers as well.

**Reply by pratapkuchipudi** [reply]
> I'm afraid I cannot accept any project that is not publicly shareable. It may lead to legal issues for not just us, but potentially our respective employers as well.

Got it

in the discussion [1814 ] mentioned that
 If you're still facing this issue, can you share a sample project for us to replicate this?

**Reply by sriv** [reply]
That is correct. A sample project is expected to be public. Ex, you upload a project with an issue to github, and we can see what's wrong.

**Comment by pratapkuchipudi** [comment]
Hi Everyone
since last one week trying with all possible solutions still issue not yet solved
here is the sample project which is to login into gmail
pls check and let me know where i am missing
thanks for your support

https://github.com/pratapkuchipudi/forbugcheck/blob/main/Automation-1.7z

**Reply by zabil** [reply]
Can you please share this as a git repository with the files unzipped and version controlled. Not everyone uses 7z plus it will be easier to point to line of code to suggest fixes.

**Reply by pratapkuchipudi** [reply]
uploaded in git
https://github.com/pratapkuchipudi/forbugcheck

it is a public one

**Reply by sriv** [reply]
I am not able to access the project https://github.com/pratapkuchipudi/forbugcheck @pratapkuchipudi 

It says 404 - Page not found. Can you please check if it is a public repo? Or you can add me as a collaborator.

**Reply by pratapkuchipudi** [reply]
Please consider this 
https://github.com/pratapkuchipudi/sampleproject

**Comment by pratapkuchipudi** [comment]
Please consider this
https://github.com/pratapkuchipudi/sampleproject

**Comment by sriv** [comment]
@pratapkuchipudi I noticed the below things in your sample:

1. Gauge.CSharp.Lib reference is old (0.7.2) - https://github.com/pratapkuchipudi/sampleproject/blob/master/netcore-template.csproj#L7. This needs to be `0.8.1`
2. The project's target framework needs to be updated to `netstandard2.1`
3. Note `trail.spec` : [Line 14](https://github.com/pratapkuchipudi/sampleproject/blob/master/specs/trail.spec#L14) and [Line 15](https://github.com/pratapkuchipudi/sampleproject/blob/master/specs/trail.spec#L15) are both essentially the same step since they have the same signature. The name of the parameter does not matter. I changed it to 

```
...
* Enter username <username> 
* Enter password <password>
...
```
And correspondingly updated the `MainAppStep.cs` to have the respective step text in the attributes:

```
...
      [Step("Enter username <username>")]
      public void EnterAuth0Username(String username) {
         Login_Decisions.EnterEmail(_driver, username);
         //Environment.SetEnvironmentVariable("username", username);
      }

      [Step("Enter password <password>")]
      public void EnteAuth0Password(String password) {
         Login_Decisions.EnterPassword(_driver, password);
      }
...
```
4. In [`trail.spec`](https://github.com/pratapkuchipudi/sampleproject/blob/master/specs/trail.spec#L6), you need not put absolute path, just `table: NRQ_Data.csv` is enough, gauge will read the file relative to the project root.

With the above changes I could see the project execute fine (I did get an error about chromedriver, but that's because I am on linux, you have the windows version of chromedriver in the project).

Hope this helps.

**Reply by sriv** [reply]
See https://github.com/pratapkuchipudi/sampleproject/pull/1 with proposed changes

**Comment by 16mohan** [comment]
Thanks Sriv.. for your quick response
taken all you changes and updated with latest chrome driver

Failed Step: Initialize webdriver
        Specification: c:\sampleproject-master\sampleproject-master\specs\trail.spec:9
        Error Message: The chromedriver.exe file does not exist in the current directory or in a directory on the PATH environment variable. The driver can be downloaded at http://chromedriver.storage.googleapis.com/index.html.
        Stacktrace: 
           at OpenQA.Selenium.DriverService.FindDriverServiceExecutable(String executableName, Uri downloadUrl)
           at OpenQA.Selenium.Chrome.ChromeDriverService.CreateDefaultService()
           at OpenQA.Selenium.Chrome.ChromeDriver..ctor(ChromeOptions options)
           at OpenQA.Selenium.Chrome.ChromeDriver..ctor()
           at netcore.template.DriverFactory.GetDriver(String browser) in c:\sampleproject-master\sampleproject-master\DriverFactory.cs:line 23
           at netcore.template.MainAppSteps.InitializeDriver() in c:\sampleproject-master\sampleproject-master\specs\MainAppSteps.cs:line 

attached the snap shot
![01](https://user-images.githubusercontent.com/51143039/103625767-70ee0080-4f61-11eb-852c-4ae4e7befd9f.GIF)

**Reply by zabil** [reply]
This is unrelated to Gauge and more to do with installing chromdriver to the path. A quick google point to several solutions for example 
https://stackoverflow.com/questions/9088367/chromedriver-does-not-exist-in-selenium-webdriver-c-sharp-test-script

There are better chances of finding help in the selenium forums.
